import-areas/import-areas.go

```diff
@@ -0,0 +1,204 @@
+}
+
+var areaTypeAndCode = map[string]string{
+	"E92": "Country",
+	"W92": "Country",
+	"E12": "Region",
+	"E06": "Unitary Authorities",
+	"W06": "Unitary Authorities",
+	"E47": "Combined Authorities",
+	"E11": "Metropolitan Counties",
+	"E10": "Counties",
+	"E09": "London Boroughs",
+	"E08": "Metropolitan Districts",
+	"E07": "Non-metropolitan Districts",
+	"E05": "Electoral Wards",
+	"W05": "Electoral Wards",
+}
+
+func lookupAreaType(code string) (string, error) {
+	prefix := code[:3]
+
+	value, found := areaTypeAndCode[prefix]
+	if !found {
+		return "", errors.New("unable to find an area type for this code")
+	}
+	return value, nil
+}
+
+func buildCountryArea(as AreaStruct) (*models.Area, error) {
+	ctx := context.Background()
+
+	area := &models.Area{}
+	for _, binding := range as.Results.Bindings {
+		area.ID = binding.AreaCode.Value
+		area.Name = binding.AreaName.Value
+		areaType, err := lookupAreaType((binding.AreaCode.Value))
+		logData := log.Data{"AreaCode": binding.AreaCode.Value}
+		if err != nil {
+			log.Event(ctx, "error returned from areaType lookup", log.ERROR, log.Error(err), logData)
+			return nil, err
+		}
+		area.Type = areaType
+		childType, err := lookupAreaType(binding.Code.Value)
+		logData = log.Data{"ChildAreaCode": binding.Code.Value}
+		if err != nil {
+			log.Event(ctx, "error returned from areaType lookup", log.ERROR, log.Error(err), logData)
+			return nil, err
+		}
+		child := &models.LinkedAreas{
+			ID:   binding.Code.Value,
+			Name: binding.Name.Value,
+			Type: childType,
+		}
+		area.ChildAreas = append(area.ChildAreas, *child)
+	}
+	return area, nil
+}
+
+func postCountryQuery(query string) (*models.Area, error) {
+	ctx := context.Background()
+
+	v := url.Values{}
+	v.Set("query", query)
+
+	client := &http.Client{}
+	req, _ := http.NewRequest("POST", "http://statistics.data.gov.uk/sparql", strings.NewReader(v.Encode()))
+	req.Header.Add("Accept", "application/sparql-results+json")
+	resp, _ := client.Do(req)
+
+	data, _ := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	var returnedCountryData AreaStruct
+
+	if err := json.Unmarshal(data, &returnedCountryData); err != nil {
+		logData := log.Data{"json file": returnedCountryData}
+		log.Event(ctx, "failed to unmarshal json", log.ERROR, log.Error(err), logData)
+		return nil, err
+	}
+
+	countryData, err := buildCountryArea(returnedCountryData)
+	if err != nil {
+		log.Event(ctx, "error returned from building the country area", log.ERROR, log.Error(err))
+		return nil, err
+	}
+
+	return countryData, nil
+}
+
+func main() {
+
+	var (
+		mongoURL string
+	)
+
+	flag.StringVar(&mongoURL, "mongo-url", "localhost:27017", "mongoDB URL")
+	flag.Parse()
+
+	ctx := context.Background()
+
+	session, err := mgo.Dial(mongoURL)
+	if err != nil {
+		log.Event(ctx, "unable to create mongo session", log.ERROR, log.Error(err))
+		os.Exit(1)
+	}
+	defer session.Close()
+
+	queryCountryEngland := `PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
```

Comment:
just a suggestion - It would be nice to group all the queries together in a var at the beginning of the file as it would be easier to read